pwa/src/templates/organizationsTemplate/OrganizationsTemplate.tsx

```diff
@@ -0,0 +1,96 @@
+  const [marginPagesDisplayed, setMarginPageDisplayed] = React.useState<number>(3);
+
+  const queryClient = new QueryClient();
+  const _useOrganisation = useOrganization(queryClient);
+  const getOrganisations = _useOrganisation.getAll({ ...filters, organizationsResultDisplayLayout: "cards" });
+
+  React.useEffect(() => {
+    if (getOrganisations.isSuccess && screenSize === "mobile") {
+      setMarginPageDisplayed(2);
+    }
+    if (getOrganisations.isSuccess && screenSize === "mobile" && getOrganisations.data.pages > 100) {
+      setMarginPageDisplayed(1);
+    }
+    if (getOrganisations.isSuccess && screenSize !== "mobile") {
+      setMarginPageDisplayed(3);
+    }
+  }, [getOrganisations]);
+
+  return (
+    <Container layoutClassName={styles.container}>
+      <div className={styles.header}>
+        <div>
+          <Heading level={2} className={styles.title}>
+            Organizations
+          </Heading>
+        </div>
+
+        <ResultsDisplaySwitch resultsDisplayType="organizationsResultDisplayLayout" />
+      </div>
+
+      <div className={styles.filtersAndResultsContainer}>
+        <div className={styles.results}>
+          <OrganizationHorizontalFiltersTemplate />
+
+          {getOrganisations.data?.results?.length === 0 && !getOrganisations.isLoading && (
+            <span>{t("No organizations found")}</span>
+          )}
+
+          {getOrganisations.data?.results && getOrganisations.data?.results?.length > 0 && (
+            <>
+              <OrganizationDisplayTemplate
+                organizations={getOrganisations.data.results}
+                type={filters.organizationsResultDisplayLayout}
+              />
+
+              {getOrganisations.data.results.length && (
+                <PaginatedItems
+                  pages={getOrganisations.data.pages}
+                  currentPage={getOrganisations.data.page}
+                  setPage={(page) => setFilters({ ...filters, organizationCurrentPage: page })}
+                  pageRangeDisplayed={2}
+                  marginPagesDisplayed={marginPagesDisplayed}
+                  containerClassName={styles.paginationContainer}
+                  pageClassName={
+                    getOrganisations.data.pages > 1000 ? styles.paginationLinkSmall : styles.paginationLink
+                  }
+                  previousClassName={
+                    getOrganisations.data.pages > 1000 ? styles.paginationLinkSmall : styles.paginationLink
+                  }
+                  nextClassName={
+                    getOrganisations.data.pages > 1000 ? styles.paginationLinkSmall : styles.paginationLink
+                  }
+                  activeClassName={
+                    getOrganisations.data.pages > 1000 ? styles.paginationActivePageSmall : styles.paginationActivePage
+                  }
+                  disabledClassName={styles.paginationDisabled}
+                  breakClassName={styles.breakLink}
```

Comment:
Do we actually need all those classNames being passed to `PaginatedItems`? 

I would expect `PaginatedItems` to have the ability to set them in its own context.

E.g.:

```
nextClassName={
  getOrganisations.data.pages > 1000 ? styles.paginationLinkSmall : styles.paginationLink
}
```

Everything's based on `.data.pages`, but we're also passing this to the component: `pages={getOrganisations.data.pages}`

Reply:
We're doing the same everywhere we're implementing this Pagination. I think the implementation is more complicated than it needs to be.